core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/store.go

```diff
@@ -34,39 +45,52 @@ var (
 
 // upkeepStateRecord is a record that we save in a local cache.
 type upkeepStateRecord struct {
-	workID string
-	state  ocr2keepers.UpkeepState
-	block  uint64
+	WorkID          common.Hash
```

Comment:
Revert to string, otherwise we will need to do multiple conversions to `common.Hash` (`workID` is represented as string across the system)

Reply:
It's fine to have this as `common.Hash` here because it's evm-specific code.